core/src/main/java/io/aiven/kafka/tieredstorage/config/RemoteStorageManagerConfig.java

```diff
@@ -95,6 +97,11 @@ public class RemoteStorageManagerConfig extends AbstractConfig {
         + "Remote Log Segment metadata on Remote Log Metadata Manager back-end. "
         + "Allowed values: " + Arrays.toString(SegmentCustomMetadataField.names());
 
+    private static final String UPLOAD_RATE_LIMIT_BYTES_CONFIG = "upload.rate.limit.bytes.per.second";
```

Comment:
When I was writing this PR, I actually wanted to read the configuration of `remote.log.manager.copy.max.bytes.per.second`. However, I found that I could only read the configuration under the prefix of `rsm.config.`, which made me worry that the rate limiting feature of Kafka 3.8 would conflict with it in the future. If we can read the configuration of `remote.log.manager.copy.max.bytes.per.second`, then we only need to remove the relevant rate limiting implementation after Kafka 3.8, and users can basically switch smoothly between the two. If we can't do this, users need to pay attention to two rate control configuration items when upgrading. At present, I can't think of a solution to this problem.

Reply:
Thanks @funky-eyes. The goal of this feature is not to replace or workaround Tiered Storage quotas. Hence the configurations are not mean to be replaceable. 
Quotas aim to limit the number of segments uploaded by counting the segments per second. This doesn't help on how fast those segments are uploaded. To have a limited speed for the upload is where I think this feature can be useful.
At the end it should be possible to use it alongside Quotas.

Reply:
I think for the customer who has fixed segment size, rate limiting on the number of segments vs rate limiting on the number of bytes for upload probably would be equivalent.  But for Aiven, their customers might have different segment sizes, more precise rate limiting control might be better.  I am fine with keeping two rate limitings.